dataikuapi/dss/ml.py

```diff
@@ -722,6 +722,86 @@ def get_partial_dependencies(self):
         return DSSPartialDependencies(data)
 
 
+class DSSSubpopulationModality(DSSExtendableDict):
+    """
+    Object to read details of a subpopulation analysis modality
+
+    Do not create this object directly, use :meth:`DSSSubpopulationAnalysis.get_modality(definition)` instead
+    """
+
+    def __init__(self, computed_as_type, data):
+        super(DSSSubpopulationModality, self).__init__(data)
+
+        if computed_as_type == "CATEGORY":
+            self.definition = DSSSubpopulationCategoryModalityDefinition(data)
+        elif computed_as_type == "NUMERIC":
+            self.definition = DSSSubpopulationNumericModalityDefinition(data)
+    
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the subpopulation analysis modality
+        """
+        return self.internal_dict
+    
+    def get_definition(self):
+        """
+        Gets the definition of the subpopulation analysis modality
+
+        :returns: definition
+        :rtype: :class:`dataikuapi.dss.ml.DSSSubpopulationModalityDefinition`
+        """
+        return self.definition
+    
+    def excluded(self):
+        """
+        Whether modality has been excluded from analysis (e.g. too few rows in the subpopulation)
+        """
+        return self.get("excluded")
+
+    def get_perf(self):
+        """
+        Gets the performance of the modality
+        """
+        if self.excluded():
+            raise ValueError("Excluded modalities do not have perf")
+        return self.get("perf")
+
+
+class DSSSubpopulationModalityDefinition(object):
+
+    def __init__(self, data):
+        self.missing_values = data.get("missing_values")
+        self.index = data.get("index")
+    
+    def missing_values(self):
```

Comment:
Consider clearer version:

```suggestion
    def is_missing_values(self):
```

How is it done elsewhere?